src/Analyzer/QueryAnalysisPass.cpp

```diff
@@ -0,0 +1,5336 @@
+  *
+  * 6. If it is table identifier lookup try to lookup identifier in current scope CTEs.
+  *
+  * 7. If identifier is not resolved in current scope, try to resolve it in parent scopes.
+  * 8. If identifier is not resolved from parent scopes and it is table identifier lookup try to lookup identifier
+  * in database catalog.
+  *
+  * Same is not done for functions because function resolution is more complex, and in case of aggregate functions requires not only name
+  * but also argument types, it is responsibility of resolve function method to handle resolution of function name.
+  *
+  * 9. If identifier was not resolved remove it from identifier_lookup_to_resolve_status table.
+  *
+  * It is okay for identifier to be not resolved, in case we want first try to lookup identifier in one context,
+  * then if there is no identifier in this context, try to lookup in another context.
+  * Example: Try to lookup identifier as function, if it is not found lookup as expression.
+  * Example: Try to lookup identifier as expression, if it is not found lookup as table.
+  */
+IdentifierResolveResult QueryAnalyzer::tryResolveIdentifier(const IdentifierLookup & identifier_lookup, IdentifierResolveScope & scope, IdentifierResolveSettings identifier_resolve_settings)
+{
+    auto it = scope.identifier_lookup_to_result.find(identifier_lookup);
+    if (it != scope.identifier_lookup_to_result.end())
+    {
+        if (!it->second.resolved_identifier)
+            throw Exception(ErrorCodes::CYCLIC_ALIASES,
+                "Cyclic aliases for identifier {}. In scope {}",
+                identifier_lookup.identifier.getFullName(),
+                scope.scope_node->formatASTForErrorMessage());
+
+        if (scope.use_identifier_lookup_to_result_cache && !scope.non_cached_identifier_lookups_during_expression_resolve.contains(identifier_lookup))
+            return it->second;
+    }
+
+    auto [insert_it, _] = scope.identifier_lookup_to_result.insert({identifier_lookup, IdentifierResolveResult()});
+    it = insert_it;
+
+    /// Resolve identifier from current scope
+
+    IdentifierResolveResult resolve_result;
+    resolve_result.resolved_identifier = tryResolveIdentifierFromExpressionArguments(identifier_lookup, scope);
+    if (resolve_result.resolved_identifier)
+        resolve_result.resolve_place = IdentifierResolvePlace::EXPRESSION_ARGUMENTS;
+
+    if (!resolve_result.resolved_identifier)
+    {
+        bool prefer_column_name_to_alias = context->getSettingsRef().prefer_column_name_to_alias;
+
+        if (unlikely(prefer_column_name_to_alias))
+        {
+            if (identifier_resolve_settings.allow_to_check_join_tree)
+            {
+                resolve_result.resolved_identifier = tryResolveIdentifierFromJoinTree(identifier_lookup, scope);
+
+                if (resolve_result.resolved_identifier)
+                    resolve_result.resolve_place = IdentifierResolvePlace::JOIN_TREE;
```

Comment:
`tryResolveIdentifierFrom*` does not have access to resolve `IdentifierResolveResult`, and probably they should not have it, because it could potentially lead to bugs, for example some of `tryResolveIdentifierFrom*` functions will set wrong resolve place.

Reply:
Another possible motivation is that now `resolve_place` is set from single place in analyzer. Easier to track.